recipes/ethash/all/conanfile.py

```diff
@@ -0,0 +1,94 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+from conan.tools.files import get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.51.1"
+
+
+class EthashConan(ConanFile):
+    name = "ethash"
+    description = "C/C++ implementation of Ethash and ProgPoW – the Ethereum Proof of Work algorithms"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/chfast/ethash"
+    topics = ("ethereum", "mining", "proof-of-work")
+    package_type = "library"
+    tool_requires = "cmake/[>=3.16.2 <4]"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "11",
+            "clang": "13",
+            "apple-clang": "14.1",
+        }
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("ethash does not support windows")
```

Comment:
Do you have any reference that it does not work on Windows from the upstream? I could not find some valid argument in the upstream: https://github.com/search?q=repo%3Achfast%2Fethash%20windows&type=code

Here it's noted that it can be built on Windows: https://github.com/chfast/ethash/issues/88